gpiozero/tones.py

```diff
@@ -61,16 +73,37 @@ class to easily represent musical tones. The class can be constructed in a
         r'(?P<semi>[%s]?)'
         r'(?P<octave>[0-9])' % ''.join(semitones.keys()))
 
-    def __new__(cls, value):
-        if isinstance(value, int):
-            if 0 <= value < 128:
-                return cls.from_midi(value)
+    def __new__(cls, value=None, **kwargs):
+        if value is None:
+            if len(kwargs) != 1:
+                raise TypeError('expected precisely one keyword argument')
+            key, value = kwargs.popitem()
+            try:
+                return {
+                    'frequency': cls.from_frequency,
+                    'midi': cls.from_midi,
+                    'note': cls.from_note,
+                }[key](value)
+            except KeyError:
+                raise TypeError('unexpected keyword argument %r' % key)
+        else:
+            if kwargs:
+                raise TypeError('cannot specify keywords with a value')
+            if isinstance(value, (int, float)):
+                if 0 <= value < 128:
+                    warnings.warn(
```

Comment:
Don't we only need the warning `if value > 0` ?

Reply:
Good point! Yup, will change that too.